pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -233,6 +233,11 @@ func endpointHealthStatus(svc *model.Service, e v1.Endpoint) model.HealthStatus
 	if e.Conditions.Ready == nil || *e.Conditions.Ready {
 		return model.Healthy
 	}
+	// For headless services if PublishNotReadyAddresses is set, we consider the endpoint healthy
+	// so that it is available in DNS queries in name table.
```

Comment:
how will this influence nds, not find it in `BuildNameTable`

Reply:
BuildNameTable just uses the services instances so there is no change needed right now.